tests/e2e/test_rosacli_policy.go

```diff
@@ -434,3 +434,364 @@ var _ = Describe("Validation testing",
 		})
 
 	})
+
+var _ = Describe("Account roles with attaching arbitrary policies",
+	labels.Feature.Policy,
+	func() {
+		defer GinkgoRecover()
+
+		var (
+			clusterID                string
+			rosaClient               *rosacli.Client
+			arbitraryPolicyService   rosacli.PolicyService
+			arbitraryPoliciesToClean []string
+			awsClient                *aws_client.AWSClient
+			err                      error
+		)
+
+		BeforeEach(func() {
+			By("Init the client")
+			rosaClient = rosacli.NewClient()
+			arbitraryPolicyService = rosaClient.Policy
+
+			awsClient, err = aws_client.CreateAWSClient("", "")
+			Expect(err).To(BeNil())
+		})
+
+		AfterEach(func() {
+			By("Clean remaining resources")
+			err := rosaClient.CleanResources(clusterID)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Delete arbitrary policies")
+			if len(arbitraryPoliciesToClean) > 0 {
+				for _, policyArn := range arbitraryPoliciesToClean {
+					err = awsClient.DeletePolicy(policyArn)
+					Expect(err).To(BeNil())
+				}
+			}
+		})
+
+		It("can be upgraded and deleted successfully - [id:74402]", labels.Critical, labels.Runtime.Day2, func() {
+			By("Prepare arbitray policies for testing")
+			awsClient, err = aws_client.CreateAWSClient("", "")
+			Expect(err).To(BeNil())
+			statement := map[string]interface{}{
+				"Effect":   "Allow",
+				"Action":   "*",
+				"Resource": "*",
+			}
+			for i := 0; i < 2; i++ {
+				arn, err := awsClient.CreatePolicy(fmt.Sprintf("ocmqe-arpolicy-%s-%d", common.GenerateRandomString(3), i), statement)
+				Expect(err).To(BeNil())
+				arbitraryPoliciesToClean = append(arbitraryPoliciesToClean, arn)
+			}
+			By("Porepare version for testing")
+			var accountRoleLowVersion string
+			versionService := rosaClient.Version
+			versionList, err := versionService.ListAndReflectVersions(rosacli.VersionChannelGroupStable, false)
+			Expect(err).To(BeNil())
+			defaultVersion := versionList.DefaultVersion()
+			Expect(defaultVersion).ToNot(BeNil())
+			lowerVersion, err := versionList.FindNearestBackwardMinorVersion(defaultVersion.Version, 1, true)
+			Expect(err).To(BeNil())
+			Expect(lowerVersion).NotTo(BeNil())
+
+			_, _, accountRoleLowVersion, err = lowerVersion.MajorMinor()
+			Expect(err).To(BeNil())
+
+			By("Create account-roles in low version")
+			ocmResourceService := rosaClient.OCMResource
+			aRolePrefix := "aroleprefix131313"
+			output, err := ocmResourceService.CreateAccountRole("--mode", "auto",
+				"--prefix", aRolePrefix,
+				"--version", accountRoleLowVersion,
+				"-y")
+			Expect(err).To(BeNil())
+			defer func() {
```

Comment:
Is it normal to use defer in tests like this, or should use an `AfterEach` to make sure resources are cleaned up?

Reply:
@ciaranRoche The common clean-up step should in AfterEach, In this case, as after the operator-roles/account roles deletion in defer , there are also checkpoint of the arbitrary policies not deleted, in other words the deletion of the created roles can be treated as a testing step, not just a common cleanup step, that's also helpful on the code readability.
In the latest commit, I have updated another defer to cleanup the testing oidc-config by moving it to AfterEach step. Thanks.